tdp/core/models/deployment_log.py

```diff
@@ -243,6 +244,70 @@ def from_operations(
                 i += 1
         return deployment_log
 
+    @staticmethod
+    def from_operations_hosts_vars(
+        collections: Collections,
+        operation_host_vars_names: list[tuple[str, Optional[str], Optional[list[str]]]],
+    ) -> "DeploymentLog":
+        """Generate a deployment plan from a list of operations, hosts and extra vars.
+
+        Args:
+            collections: Collections to retrieve the operations from.
+            operation_host_vars_names: List of operations, hosts and extra vars to perform.
+
+        Raises:
+            MissingOperationError: If an operation is a noop.
+            MissingHostForOperationError: If an operation is not found in the collections.
+        """
+        operations_list = []
+        hosts_list = []
+        vars_list = []
+        for operation_names, host_names, var_names in operation_host_vars_names:
+            operations_list.append(operation_names)
+            hosts_list.append(host_names)
+            vars_list.append(var_names)
+
+        hosts_list = list(sorted(hosts_list)) if None not in hosts_list else None
+        extra_vars_set = set(
+            list(itertools.chain(*[item for item in vars_list if item is not None]))
+        )
+        extra_vars_list = list(sorted(extra_vars_set)) if vars_list else None
+
+        deployment_log = DeploymentLog(
+            deployment_type=DeploymentTypeEnum.CUSTOM,
+            options={
+                "operations": operations_list,
+            },
+            status=DeploymentStateEnum.PLANNED,
+        )
+        if hosts_list is not None:
+            deployment_log.options["hosts"] = hosts_list
+        if extra_vars_list is not None:
+            deployment_log.options["extra_vars"] = extra_vars_list
```

Comment:
IMO this whole part can be replaced by the following:

```suggestion
        deployment_log = DeploymentLog(
            deployment_type=DeploymentTypeEnum.CUSTOM,
            status=DeploymentStateEnum.PLANNED,
        )
```

DeploymentLog.options will be used to reproduce a past deployment.

In the case of a `CUSTOM` deployment, those options can't help us because each operation may have its own host and extra vars. Hence, we won't be able to generate the same deployment plan from what you saved.

Reply:
Done, I removed the use of ```options``` and the associated code

Reply:
Re-done, I remove all the useless code ^^'